mule-http-connector/src/main/java/org/mule/extension/http/internal/request/HttpRequester.java

```diff
@@ -262,7 +262,7 @@ private void doRequestWithRetry(HttpExtensionClient client, HttpRequesterConfig
             }
 
             logger.error(getErrorMessage(httpRequest));
-            HttpError error = exception instanceof TimeoutException ? TIMEOUT : CONNECTIVITY;
+            HttpError error = exception instanceof TimeoutException ? REQUEST_TIMEOUT : CONNECTIVITY;
```

Comment:
As discussed, Currently we are throwing HTTP:Timeout in case of no response, after this change we will throw HTTP:REQUEST_TIMEOUT. 
As per earlier implementation customers might be expecting HTTP:Timeout here so this might break few things on their side. It would suggest to keep it same.
For the scenario where status code is 408, we can try if HTTP:Timeout can be thrown there as well to make sure it is same everywhere.
If above is not possible then can go ahead with REQUEST_TIMEOUT at both places and need to update the documentation as well.